sw/device/lib/crypto/drivers/hmac.h

```diff
@@ -112,14 +129,14 @@ status_t hmac_hash_sha512(const uint8_t *msg, size_t msg_len, uint32_t *digest);
  * The key should be pre-processed into a buffer the size of a full message
  * block, according to FIPS 198-1, section 4.
  *
- * @param key_block Input key block (`kHmacSha256BlockWords` words).
+ * @param key HMAC key.
```

Comment:
I think this function probably needs to say something explicit about the fact that it only looks at `kHmacSha256BlockWords` words of the key (no matter what `key->key_len` contains).

Similarly for the other fixed-size functions below.